histoqc/__main__.py

```diff
@@ -76,7 +76,12 @@ def main(argv=None):
         lm.logger.warning(f"Configuration file not set (--config), using default")
         config.read_string(read_config_template('default'))
     else:
-        config.read(args.config)
+        try:
```

Comment:
I think you want an elif  pattern here, with an explicit check to see if the file "args.config" exists

some general explanation:
https://www.pythonforbeginners.com/control-flow-2/when-to-use-try-catch-instead-of-if-else
https://www.geeksforgeeks.org/try-except-vs-if-in-python/
https://stackoverflow.com/questions/1835756/using-try-vs-if-in-python

so one should ask how frequently the user will take this approach? and if it is frequent, an "if" is more appropriate (and faster)

note as well that this "except" creates a slightly unexpected situation. what if the file *does* exist, but it is not parseable? i.e., you're setup to catch one error but not another

Reply:
may be I did not think correctly.
I am assuming for a user to use -c config parameter, the input type will be one of the listed configurations.
```
- default:  config.ini
- clinical: config_clinical.ini
- first:    config_first.ini
- ihc:      config_ihc.ini
- light:    config_light.ini
- v2.1:     config_v2.1.ini
```
Example
```
> python -m histoqc -c v2.1 -n 3 "*.svs"
or
> python -m histoqc -c light -n 3 "*.svs"
```

Is there a possibility that the user will have a filepath given ?
Current code is incorrect, hence it will not work at all if any config parameter is provided, and to make it work 
```
config.read(args.config) line 79
needs to be changed to
 config.read_string(read_config_template(args.config))
```

Reply:
try catch will be removed completely

Reply:
Or we can just change it to 
```
    if args.config:
        if args.config.endswith('.ini'):
            config.read(args.config)
        else:
            config.read_string(read_config_template(args.config))
    else:
        lm.logger.warning(f"Configuration file not set (--config), using default")
        config.read_string(read_config_template('default'))
```

Reply:
I think we're looking for something like this:

```
if not args.config: #not set, load default
    lm.logger.warning(f"Configuration file not set (--config), using default")
    config.read_string(read_config_template('default'))
elif os.path.exists(args.config): #check to see if exists as a file with higher priority
    config.read_file(open(args.config))
else: #check to see if named template
    lm.logger.warning(f"Configuration file {args.config} assuming to be a template...checking.")
    config.read_string(read_config_template(args.config))
    
```

Reply:
Ah okay, no problem will change it and send a pull request 
again

Reply:
Please make sure to test it first, I wrote that code in notepad : )

Reply:
Yes I will